File: Unit4A1.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-11-24
// Fileoverview: This program asks the user for two odd numbers and prints all even numbers between them.

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	// variable declaration
	var startString string
	var endString string
	var startValue int
	var endValue int

	reader := bufio.NewReader(os.Stdin)

	// ask for starting value
	fmt.Print("Please enter a starting value (it must be an odd number): ")
	startString, _ = reader.ReadString('\n')
	startString = strings.TrimSpace(startString)
	startValue, _ = strconv.Atoi(startString)

	// ask for ending value
	fmt.Print("Please enter an ending value (it must be an odd number): ")
	endString, _ = reader.ReadString('\n')
	endString = strings.TrimSpace(endString)
	endValue, _ = strconv.Atoi(endString)

	fmt.Println("\nEven numbers between your values:")

	// loop between values
	for i := startValue; i <= endValue; i++ {
		if i%2 == 0 {
			fmt.Printf("%d ", i)
		}
	}

	fmt.Println("\nDone.")
}

File: Unit4A1.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-24
 * @fileoverview This program prints all the even numbers between two odd values.
 */

// variables
let startValue: number = 0;
let endValue: number = 0;

// get starting value
const startInput = prompt("Please enter a starting value (it must be an odd number): ");
startValue = parseInt(startInput || "0");

// get ending value
const endInput = prompt("Please enter an ending value (it must be an odd number): ");
endValue = parseInt(endInput || "0");

// print even numbers
let output = "";

// loop through values
for (let counter = startValue + 1; counter < endValue; counter++) {
  if (counter % 2 === 0) {
    // add number and a space (plain text, not template literal)
    output = output + counter + " ";
  }
}

// print final result (plain text)
console.log(output);

console.log("Done.");
